### DISCUSSION TITLE
Should `throwOnError` be enabled by default?

### DISCUSSION THREAD

**Discussion by donysukardi** [TanStack/query#1349] [Ideas]
The examples on `useMutation` always use the following pattern.
- https://react-query.tanstack.com/docs/guides/mutations#basic-mutations
- https://react-query.tanstack.com/docs/guides/mutations#mutation-variables

This will not work with the default config, i.e. `throwOnError: false`.

```js
try {
  await doMutate();
} catch(e) {
  // handle error
}
```

Should react-query enable it by default? Otherwise, perhaps it'd be better to include a note in the examples

**Comment by TkDodo** [comment]
v3 differentiates between `mutate` and `mutateAsync`: https://react-query-beta.tanstack.com/guides/mutations#promises

One will throw, the other one has callbacks.

**Reply by donysukardi** [reply]
I'm wondering if we should add a note in the current docs, in both [mutations guide](https://react-query.tanstack.com/docs/guides/mutations) and [important defaults](https://react-query.tanstack.com/docs/guides/important-defaults).

I can create a PR for that

**Reply by TkDodo** [reply]
Please note that this applies to v3, _not_ the v2 (which you are linking to). The v3 docs do cover that topic, but If you think those docs are missing something, feel free to create a PR
